android/java/org/chromium/chrome/browser/component_updater/BraveComponentUpdater.java

```diff
@@ -0,0 +1,161 @@
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import org.chromium.base.Log;
+import org.chromium.base.ThreadUtils;
+import org.chromium.base.annotations.CalledByNative;
+import org.chromium.base.annotations.JNINamespace;
+import org.chromium.base.annotations.NativeMethods;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * Class for handling progress of component update in Java.
+ * Basically this is just bridge to native component updater classes at
+ * brave_component_updater_android.cc.
+ */
+@JNINamespace("chrome::android")
+public class BraveComponentUpdater {
+    private static final String TAG = "BCU";
+    private long mNativeBraveComponentUpdaterAndroid;
+    private static BraveComponentUpdater sBraveComponentUpdater;
+
+    public static BraveComponentUpdater get() {
+        ThreadUtils.assertOnUiThread();
+        if (sBraveComponentUpdater == null) {
+            sBraveComponentUpdater = new BraveComponentUpdater();
+        }
+        return sBraveComponentUpdater;
+    }
+
+    private BraveComponentUpdater() {
+        init();
+    }
+
+    @CalledByNative
+    private void setNativePtr(long nativePtr) {
+        assert mNativeBraveComponentUpdaterAndroid == 0;
+        mNativeBraveComponentUpdaterAndroid = nativePtr;
+    }
+
+    private void init() {
+        if (mNativeBraveComponentUpdaterAndroid == 0) {
+            BraveComponentUpdaterJni.get().init(BraveComponentUpdater.this);
+        }
+    }
+
+    @Override
+    protected void finalize() {
+        destroy();
+    }
+
+    private void destroy() {
+        if (mNativeBraveComponentUpdaterAndroid != 0) {
+            BraveComponentUpdaterJni.get().destroy(mNativeBraveComponentUpdaterAndroid);
+            mNativeBraveComponentUpdaterAndroid = 0;
+        }
+    }
+
+    /**
+     * Listener for the component update changes.
+     */
+    public interface ComponentUpdaterListener {
+        // Invoked when component update info has changed.
+        public void onComponentUpdateEvent(int event, String id);
+    }
+
+    private final List<ComponentUpdaterListener> mComponentUpdaterListeners =
+            new CopyOnWriteArrayList<ComponentUpdaterListener>();
+
+    public void addComponentUpdateEventListener(ComponentUpdaterListener listener) {
+        ThreadUtils.assertOnUiThread();
+        mComponentUpdaterListeners.add(listener);
+    }
+
+    public void removeComponentUpdateEventListener(ComponentUpdaterListener listener) {
+        ThreadUtils.assertOnUiThread();
+        mComponentUpdaterListeners.remove(listener);
+    }
+
+    @CalledByNative
+    protected void componentStateUpdated(int event, String id) {
+        for (ComponentUpdaterListener listener : mComponentUpdaterListeners) {
+            listener.onComponentUpdateEvent(event, id);
+        }
+    }
+
+    /**
+     * Class describing the progress of component download.
+     */
+    public class CrxUpdateItem {
```

Comment:
Inner class `CrxUpdateItem` may be static -> public static class CrxUpdateItem

Reply:
thanks, fixed at https://github.com/brave/brave-core/pull/19086/commits/54818d8dcc99e3f267ebe48dde6da4ba6625545f#diff-b0ad67a3cc1f46d4b3163f2b99f39d57f940215f0c4a0a327a02bea2cf99f35cR98